docs/threat_models/smart_contracts/multipool_autoswap/mpas_sequence.puml

```diff
@@ -0,0 +1,182 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+header Agoric Operating Systems
+footer Page %page% of %lastpage%
+title Multipool Auto Swap Sequence Diagram
+
+autonumber
+actor "MPAS Creator" as masc #lime
+actor "Pool Adder N" as pa #yellow
+actor "Pool User" as pu
+control "Zoe" as zoe
+database "MPAS Contract" as mpas
+
+== MPAS Creation ==
+masc -> mpas: Create instance
+return ""creatorFacet""
+== Pool Addition SML:BCK==
+autonumber 10
+pa -> mpas: ""addPool()""
```

Comment:
In order to get from creating an instance to adding a pool, the creator has to publish the instance of the contract somewhere. Otherwise, even though the addPool() method is "public", no one would be able to find it.  It is completely possible in this model to create an AMM, and add pools, and only tell friends about it.

Reply:
Is it possible to enumerate or directly reference this contract somehow through reflection or other means? Or does it need to be explicitly published?